sdclientapi/__init__.py

```diff
@@ -326,9 +326,12 @@ def get_submission(self, submission: Submission) -> Submission:
         :param submission: Submission object we want to update.
         :returns: Updated submission object from the server.
         """
-        path_query = "api/v1/sources/{}/submissions/{}".format(
-            submission.source_uuid, submission.uuid
-        )
+        if submission.download_url != "":
+            path_query = submission.download_url
+        else:
+            path_query = "api/v1/sources/{}/submissions/{}".format(
+                submission.source_uuid, submission.uuid
+            )
```

Comment:
if we allow `Submission` objects that only have the `download_url` attribute defined, then we need to remove references to other attributes on that object in case they are empty strings, i.e. `submission.uuid` in the `WrongUUIDError` that is thrown for 404s 

(reasonable to just show the URL or the UUID in the exception depending on which is defined so it's clear to the user what happened)